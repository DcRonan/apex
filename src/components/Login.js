import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import { fetchUser } from '../actions';

const Login = () => {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const dispatch = useDispatch();

  const onSubmit = e => {
    e.preventDefault();

    const userInfo = {
      username,
      password,
    };

    if (username === '' || password === '') return toast.error('username or password blank');
    return dispatch(fetchUser(userInfo));
  };

  return (
    <div>
      <h1>Login Form</h1>
      <form onSubmit={onSubmit}>
        <input
          type="text"
          name="username"
          placeholder="Username"
          value={username}
          onChange={e => setUsername(e.target.value)}
        />
        <br />
        <input
          type="password"
          name="password"
          placeholder="Password"
          value={password}
          onChange={e => setPassword(e.target.value)}
        />
        <br />
        <input type="submit" value="Login" />
      </form>
    </div>
  );
};

export default Login;
