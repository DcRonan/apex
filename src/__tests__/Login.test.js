import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Login from '../components/Login';
import store from '../store';

describe('checks if Home component renders correctly', () => {
  test('should renders correct text', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>,
    );
    const text = screen.getByText('APEX');
    expect(text).toBeInTheDocument();
  });

  test('should render correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Login />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
