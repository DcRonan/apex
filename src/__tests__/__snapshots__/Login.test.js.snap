// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if Home component renders correctly should render correctly 1`] = `
<div
  className="h-screen flex flex-col justify-center items-center bg-lightblue"
>
  <div
    className="mb-10"
  >
    <h1
      className="font-nimbusbold text-4xl italic text-gray-700"
    >
      APEX
    </h1>
  </div>
  <div
    className="px-16 w-full"
  >
    <form
      className="flex flex-col space-y-3 font-nimbuslight"
      onSubmit={[Function]}
    >
      <input
        className="px-3 py-2 focus:border-green-500 focus:outline-none"
        name="username"
        onChange={[Function]}
        placeholder="Username"
        type="text"
        value=""
      />
      <input
        className="px-3 py-2 focus:border-4 focus:border-green-500 focus:outline-none"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <input
        className="px-3 py-2 bg-green-500 text-white border-0 focus:outline-none"
        type="submit"
        value="Login"
      />
    </form>
  </div>
</div>
`;
